---
title: "methylQC Report"
output: html_document
params:
  data: NA
  cytosines: NA
  depth: NA
---

```{r eval = FALSE, echo = FALSE}
# For PDF output, change the header to have "output: pdf_document".
#
# Note that due to an issue in rmarkdown, the default value of a parameter in
# the header cannot be `NULL`, so I used a default of `NA` for the default value
# of `n`.
```

```{r eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
library(ggplot2)
library(dplyr)
library(methylQC)
```


# Coverage  

## Genome browser view

`r input$chrom` between `r rv$start` and `r rv$end`

```{r browser, echo=FALSE, message=FALSE, warning=FALSE}
plotBrowser(rv$data, start = rv$start, stop = rv$end)
```

## Sequencing depth

```{r depth, echo=FALSE, message=FALSE, warning=FALSE}
hist(params$depth, col = "orangered3",
         main = paste("Sequencing depth for ", as.character(input$chrom)), xlab = "Depth")
```

```{r, echo=FALSE, message=FALSE, warning=FALSE}
methylomeStats(params$data) %>%
  knitr::kable()
```

## Covered cytosines

```{r cytosines, echo=FALSE, message=FALSE, warning=FALSE}
survival <- coverageSurvival(rv$data, cytosines = params$cytosines)
plotCoverage(survival)
```

```{r coverage, echo=FALSE, message=FALSE, warning=FALSE}
cc <- dplyr::filter(survival, depth %in% c(1, 5, 10))
tidyr::spread(cc, depth, Cytosines) %>% knitr::kable()
```

# Non-conversion rate

## Lambda genome coverage

```{r lambda, echo=FALSE, message=FALSE, warning=FALSE}
l <- subset(params$data, params$data$chr == "L")
ggplot(l, aes(depth)) + geom_histogram(fill = "darkgreen", color="black") +
      theme_bw() + ggtitle("Lambda chromosome sequencing depth")
```

## Non-conversion rates

```{r conversion, echo=FALSE, message=FALSE, warning=FALSE}
nonConversion(l) %>%
  knitr::kable()
```

